src/rpc/request.cpp

```diff
@@ -142,6 +149,9 @@ bool GetAuthCookie(std::string *cookie_out)
     std::ifstream file;
     std::string cookie;
     fs::path filepath = GetAuthCookieFile();
+    if (filepath.empty()) {
+        return true; // -norpccookiefile was specified
```

Comment:
Do I understand correctly that we're basically treating an empty path as an `std::nullopt` here (and commenting it everywhere since we admit it's hard to understand)?
I understand that we can't fix everything in the same PR, but would like us to consider if it still makes sense to build on previous hacks.